README.md

```diff
@@ -1,121 +1,26 @@
-# dig: Dependency Injection Framework for Go
+# :hammer: dig [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov] [![Report Card][report-card-img]][report-card]
 
-[![GoDoc][doc-img]][doc]
-[![Coverage Status][cov-img]][cov]
-[![Build Status][ci-img]][ci]
-[![Report Card][report-card-img]][report-card]
+A dependency injection toolkit for Go.
```

Comment:
"reflection based" might be good to shove somewhere in there to hint at the "bad for" section you indicated